smart-contracts/contracts/Oracle.sol

```diff
@@ -125,52 +91,21 @@ contract Oracle is OracleStorage {
         return valid;
     }
 
-    /*
-     * @dev: checkBridgeProphecy
-     *       Operator accessor method which checks if a prophecy has passed
-     *       the validity threshold, without actually completing the prophecy.
-     */
-    function checkBridgeProphecy(uint256 _prophecyID)
-        public
-        view
-        onlyOperator
-        isPending(_prophecyID)
-        returns (bool, uint256, uint256)
-    {
-        require(
-            CosmosBridge(cosmosBridge).isProphecyClaimActive(_prophecyID) == true,
-            "Can only check active prophecies"
-        );
-        return getProphecyThreshold(_prophecyID);
-    }
-
     /*
      * @dev: processProphecy
      *       Calculates the status of a prophecy. The claim is considered valid if the
      *       combined active signatory validator powers pass the consensus threshold.
      *       The threshold is x% of Total power, where x is the consensusThreshold param.
      */
     function getProphecyThreshold(uint256 _prophecyID)
-        internal
+        public
         view
         returns (bool, uint256, uint256)
     {
         uint256 signedPower = 0;
-        uint256 totalPower = Valset(valset).totalPower();
-
-        // Iterate over the signatory addresses
-        for (
-            uint256 i = 0;
-            i < oracleClaimValidators[_prophecyID].length;
-            i = i.add(1)
-        ) {
-            address signer = oracleClaimValidators[_prophecyID][i];
-
-            // Only add the power of active validators
-            if (Valset(valset).isActiveValidator(signer)) {
-                signedPower = signedPower.add(Valset(valset).getValidatorPower(signer));
-            }
-        }
+        uint256 totalPower = totalPower;
+
+        signedPower = oracleClaimValidators[_prophecyID];
 
         // Prophecy must reach total signed power % threshold in order to pass consensus
         uint256 prophecyPowerThreshold = totalPower.mul(consensusThreshold);
```

Comment:
we can save it in contract, don't need compute it each time.

Reply:
We do have to compute if we are over the threshold when we call this function. If we store it in the contract, that is more storage and it costs more gas.

Reply:
We do store the power that has signed off in the smart contract.

Reply:
Right here: https://github.com/Sifchain/sifnode/pull/442/files#diff-d271781e15da4ca1fd42b79c262e4e6f33623521edb6a96b361dd905335011dcR80